libbpfgo/libbpfgo.go

```diff
@@ -598,14 +623,80 @@ func doAttachKprobeLegacy(prog *BPFProg, kp string, isKretprobe bool) (*BPFLink,
 var eventChannels = make(map[uintptr]chan []byte)
 var lostChannels = make(map[uintptr]chan uint64)
 
+func (m *Module) InitRingBuf(mapName string, eventsChan chan []byte) (*RingBuffer, error) {
+	bpfMap, err := m.GetMap(mapName)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := uintptr(bpfMap.fd)
+	if eventsChan == nil {
+		return nil, fmt.Errorf("events channel can not be nil")
+	}
+	eventChannels[ctx] = eventsChan
+
+	rb := C.init_ring_buf(bpfMap.fd)
+	if rb == nil {
+		return nil, fmt.Errorf("")
+	}
+
+	ringBuf := &RingBuffer{
+		rb:     rb,
+		bpfMap: bpfMap,
+		stop:   make(chan bool),
+	}
+	m.ringBufs = append(m.ringBufs, ringBuf)
+	return ringBuf, nil
+}
+
+func (rb *RingBuffer) Start() {
+	go rb.poll()
+}
+
+func (rb *RingBuffer) Stop() {
+	rb.stop <- true
+}
+
+func (rb *RingBuffer) Close() {
```

Comment:
good catch, thinking about that what happens if Close is called before Stop. In this case you chose to wait until the buffer is closed, but I think an easier and reasonable approach, is to just call Stop? either way I don't think close should block the caller

Reply:
also, whatever we decide here probably applies to perf buffer (but can be implemented in a separate PR) cc: @yanivagman

Reply:
ok after posting the review I see that this was already discussed so I'll take a moment to re-read the history first